src/DotNetWorker/Diagnostics/WorkerMessage.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    internal static class WorkerMessage
+    {
+        private static readonly AsyncLocal<bool> _isSystemLog = new AsyncLocal<bool>();
+
+        internal static bool IsSystemLog => _isSystemLog.Value;
+
+        public static Action<ILogger, Exception?> Define(LogLevel logLevel, EventId eventId, string formatString)
```

Comment:
The new approach to making sure that we are flagging worker (system) logs internally:
1. Do not use `logger.Log____`. Instead, always use `WorkerMessage.Define` for logs, which is a wrapper around `LoggerMessage.Define`.
2. I want to follow up with an analyzer to make sure we're doing this everywhere in the future (we currently have almost no system logging).
3. If this gets missed -- the log will still be written, but it won't be treated as a system log.
4. Yes, it's a tiny little bit extra work on our side -- but it takes care of the issues I had with the other approach.
5. If we need to expand the overloads for `WorkerMessage` with more parameters... we can just add them here as needed.